clients/python-wrapper/lakefs/client.py

```diff
@@ -243,21 +241,32 @@ def from_aws_role(
     :param kwargs: The arguments to pass to the client.
     :return: A lakeFS client.
     """
-
     client = Client(**kwargs)
     lakefs_host = urlparse(client.config.host).hostname
     identity_token = _get_identity_token(session, lakefs_host, presign_expiry=presigned_ttl,
                                          additional_headers=additional_headers)
     external_login_information = ExternalLoginInformation(token_expiration_duration=ttl_seconds, identity_request={
         "identity_token": identity_token
     })
-
     with api_exception_handler():
         auth_token = client.sdk_client.auth_api.external_principal_login(external_login_information)
-
+    os.environ[_LAKECTL_CREDENTIALS_ACCESS_TOKEN] = auth_token.token
     client.config.access_token = auth_token.token
     return client
 
+def _check_for_host():
```

Comment:
In this case I would rather just move the entire logic to the ServerConfiguration and create the client twice